src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -16,141 +16,172 @@
+    }
+    proprietaireCreationDTO.setNumTel(numTel);
+
+    StringBuilder generatedPassword = new StringBuilder(16);
+    String allowedchar = "abcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()_+-=[]?";
+    SecureRandom random = new SecureRandom();
+    for (int i = 0; i < 16; i++) {
+      int index = random.nextInt(allowedchar.length());
+      generatedPassword.append(allowedchar.charAt(index));
+    }
 
-        String numTel = proprietaireCreationDTO.getNumTel();
-        numTel = numTel.replaceAll(" ", "");
-        if (proprietaireRepository.findFirstByNumTel(numTel).isPresent()){
-            throw new ProprietaireNumTelAlreadyExistException();
-        }
-        proprietaireCreationDTO.setNumTel(numTel);
+    System.out.println(generatedPassword); // waiting for SMTP
 
-        String hashedPassword = hashPassword(proprietaireCreationDTO.getMotDePasse());
+    Proprietaire proprietaire = ProprietaireMapper.creationDtoToEntity(proprietaireCreationDTO,
+        hashPassword(generatedPassword.toString()));
+    return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+  }
 
-        proprietaireCreationDTO.setMotDePasse(hashedPassword);
+  public List<ProprietaireDTO> getProprietaire(boolean isLogement) {
+    return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+  }
 
-        Proprietaire proprietaire = ProprietaireMapper.creationDtoToEntity(proprietaireCreationDTO);
+  public ProprietaireDTO getProprietaireById(@NonNull Long id, boolean isLogement) {
 
-        return ProprietaireMapper.entityToDto(proprietaireRepository.save(proprietaire), false);
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
     }
+    return ProprietaireMapper.entityToDto(optionalProprietaire.get(), isLogement);
+  }
 
-    public List<ProprietaireDTO> getProprietaire(boolean isLogement){
-        return ProprietaireMapper.entityListToDtoList(proprietaireRepository.findAll(), isLogement);
+  public ProprietaireDTO getProprietaireByEmail(@NonNull String email) {
+    Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findFirstByEmail(email);
+    if (optionalProprietaire.isEmpty()) {
+      throw new ProprietaireNotFoundException();
     }
 
-    public ProprietaireDTO getProprietaireById(@NonNull Long id, boolean isLogement){
-
-        Optional<Proprietaire> optionalProprietaire = proprietaireRepository.findById(id);
-        if(optionalProprietaire.isEmpty()){
-            throw new ProprietaireNotFoundException();
-        }
-        return ProprietaireMapper.entityToDto(optionalProprietaire.get(), isLogement);
+    return ProprietaireMapper.entityToDto(optionalProprietaire.get(), false);
+  }
+
+  @Transactional
+  public ProprietaireDTO updateProprietaire(@NonNull Long id,
+      @NonNull ProprietaireUpdateDTO proprietaireUpdateDTO) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'proprietaireUpdateDTO' must contain no more than '1' consecutive capital letters.